### PR TITLE
Spelling utils

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift utils, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101092698

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please test

**Comment by tshortli** [comment]
```
ImportError: cannot import name 'TRIVIAS' from 'gyb_syntax_support.Trivia' (/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/swift/utils/gyb_syntax_support/Trivia.py)
```

**Comment by jsoref** [comment]
Right, I should have tagged that here. I think I called it out in the parent PR.

It's easy to drop the change, or to pull in the piece that was clipped by the directory split.

I won't be able to do either until tonight, but hopefully I'll have direction from a reviewer by then.

**Comment by tshortli** [comment]
> Right, I should have tagged that here. I think I called it out in the parent PR.
> 
> It's easy to drop the change, or to pull in the piece that was clipped by the directory split.
> 
> I won't be able to do either until tonight, but hopefully I'll have direction from a reviewer by then.

I think the most straightforward thing to do here is drop the `trivia` change from this PR. Everything else looks good to me.

**Comment by rintaro** [comment]
`TRIVIAS` change requires a synchronous change in `swift-syntax` repo. I would prefer to drop the change from this PR.

As for the `TRIVIAS`, I don't have a strong opinion. Though, as a non-native English speaker, it doesn't look strange, and the presence of `S` is helpful to indicate that it is a collection. But I don't know how native speakers feel. Paging @ahoppen

**Comment by tshortli** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci python lint

File: utils/gyb_syntax_support/StmtNodes.py [hunk 1]
```diff
@@ -197,8 +197,8 @@
              Child('Condition', kind='Syntax',
                    node_choices=[
                        Child('Expression', kind='Expr'),
-                       Child('Availablity', kind='AvailabilityCondition'),
-                       Child('Unavailablity', kind='UnavailabilityCondition'),
+                       Child('Availability', kind='AvailabilityCondition'),
+                       Child('Unavailability', kind='UnavailabilityCondition'),
```
Comment (tshortli) [inline utils/gyb_syntax_support/StmtNodes.py:201]: I think these changes may be causing some skew in generated sources that need to match the SwiftSyntax repository:

```
diff --recursive --exclude '.*' --context=0 /var/folders/pf/p5g2vb2j0yx1x6vlmxdppc700000gn/T/tmpx7cx6_tm/SyntaxBuilders.swift /Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/swift-syntax/Sources/SwiftSyntax/gyb_generated/SyntaxBuilders.swift
*** /var/folders/pf/p5g2vb2j0yx1x6vlmxdppc700000gn/T/tmpx7cx6_tm/SyntaxBuilders.swift	Thu Apr 21 21:39:42 2022
--- /Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/swift-syntax/Sources/SwiftSyntax/gyb_generated/SyntaxBuilders.swift	Thu Apr 21 15:19:40 2022
***************
*** 6434,6435 ****
!   public mutating func useDeclname(_ node: DeclNameSyntax) {
!     let idx = TargetFunctionEntrySyntax.Cursor.declname.rawValue
--- 6434,6435 ----
!   public mutating func useDelcname(_ node: DeclNameSyntax) {
!     let idx = TargetFunctionEntrySyntax.Cursor.delcname.rawValue
```

The easiest path to unblocking the rest of the changes would be to revert the changes in this file, but we could also make a corresponding PR in https://github.com/apple/swift-syntax

Comment (ahoppen) [reply utils/gyb_syntax_support/StmtNodes.py:201]: Most of the other changes to `gyb_syntax_support` require corresponding changes to https://github.com/apple/swift-syntax. Could you create a matching PR in that repo by 
- Checking out swift-syntax next to swift
- Run `swift-syntax/build-script.py --toolchain doesnt-matter --degyb-only`

Alternatively, we could consider dropping all changes in `gyb_syntax_support` from this PR.

File: utils/gyb_syntax_support/Trivia.py [hunk 2]
```diff
@@ -62,13 +62,13 @@ def is_collection(self):
 ]
 
 
-def verify_no_duplicate_serialization_codes(trivias):
+def verify_no_duplicate_serialization_codes(trivia):
```
Comment (ahoppen) [inline utils/gyb_syntax_support/Trivia.py:65]: I would prefer to continue using `trivias` here to match the global `TRIVIAS` variable, even though it’s not a valid English pluralization. It also works around the problem to rename `trivia` to `item` below.

File: utils/parser-lib/profile-input.swift [hunk 3]
```diff
@@ -294,7 +294,7 @@ class BrowserViewController: UIViewController {
             self.displayedPopoverController = nil
         }
 
-        // If we are displying a private tab, hide any elements in the tab that we wouldn't want shown
+        // If we are displaying a private tab, hide any elements in the tab that we wouldn't want shown
```
Comment (ahoppen) [inline utils/parser-lib/profile-input.swift:297]: I would prefer to keep this file as-is because this is a real-world Swift file that we use to make profile-guided optimizations for the SwiftSyntax parser. For that I would prefer to keep it in its original state. 

In practice it shouldn’t really matter because the changes are unlikely to affect parsing in any significant way, but still.

**Review by ahoppen — COMMENTED** [review]
Thanks a lot for this PR. There are way more spelling errors than I expected. Fixing them is great. 🙏 

I reviewed the changes in `gyb_syntax_support`, `incrparse` and `parser-lib` and added a few inline comments.

File: utils/gyb_syntax_support/StmtNodes.py (hunk 1 above)
Comment (jsoref) [reply utils/gyb_syntax_support/StmtNodes.py]: I've dropped them from this PR.

**Comment by tshortli** [comment]
@swift-ci please test

File: utils/gyb_syntax_support/StmtNodes.py (hunk 1 above)
Comment (ahoppen) [reply utils/gyb_syntax_support/StmtNodes.py]: 👍 If you want to open a separate PR for the `gyb_syntax_support` changes, I would really appreciate it. If you need help with the corresponding swift-syntax change, I’m very happy to help you out!

**Comment by tshortli** [comment]
@jsoref looks good, just needs a conflict resolution before this can land

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Review by ahoppen — APPROVED** [review]
Changes in `incrparse` look good to me – I didn’t review the other changes.

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Review by tshortli — APPROVED** [review]
